torchtext/experimental/models/roberta_model.py

```diff
@@ -0,0 +1,74 @@
+import os
+import torch
+import torch.nn as nn
+from torchtext.experimental.modules import TransformerEncoder
+from .roberta_transform import RobertaTransform
+from torchtext.utils import download_from_url, extract_archive, files_exist, load_args_from_json
+
+
+class RobertaModel(nn.Module):
+    """Contain a Roberta encoder model."""
+
+    def __init__(self, args, encoder):
+        super(RobertaModel, self).__init__()
+        self.args = args
+        self.encoder = encoder
+        self.transform = None
+
+    @classmethod
+    def build_model(self, args):
+        encoder = TransformerEncoder(args.ntoken, args.embed_dim, args.nhead,
+                                     args.feedforward_dim, args.nlayers, args.dropout)
+        return cls(args, encoder)
+
+    def load_transform(self, directory='./', tokenizer_file="sentencepiece.bpe.model", vocab_file='vocab.txt'):
+        self.transform = RobertaTransform.from_pretrained(directory=directory,
+                                                          tokenizer_file=tokenizer_file,
+                                                          vocab_file=vocab_file)
+
+    # [TODO] use torch.hub.load_state_dict_from_url
+    @classmethod
+    def from_pretrained(cls, args, directory='./', checkpoint_file="model.pt"):
```

Comment:
How about adding a "PreTrainedModel" as super class or mixin? Only a "PreTrainedModel" is required to implement this method, the others(e.g. CNN) is not.

Reply:
Yup. This func is general enough and we will move it to the base class later.